### PR TITLE
zoxide: 0.9.2 -> unstable-2023-11-20

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

latest version of nushell will break with zoxide. The fix landed on master, but the maintainer hasn't had time to release a new version (I've sent him 3 messages in the last 2 weeks).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/zoxide/default.nix [hunk 1]
```diff
@@ -10,13 +10,13 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "zoxide";
-  version = "0.9.2";
+  version = "unstable-20-11-2023";
```
Comment (cole-h) [inline pkgs/tools/misc/zoxide/default.nix:13]:
```suggestion
  version = "unstable-2023-11-20";
```

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 273882` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>felix-fm</li>
    <li>vimPlugins.telescope-zoxide</li>
    <li>vimPlugins.zoxide-vim</li>
    <li>yazi</li>
    <li>zoxide</li>
  </ul>
</details>

File: pkgs/tools/misc/zoxide/default.nix (hunk 1 above)
Comment (happysalada) [reply pkgs/tools/misc/zoxide/default.nix]: I've also added a small explanation in the commit message as well.
let me know if I've missed anything.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 273882` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>felix-fm</li>
    <li>vimPlugins.telescope-zoxide</li>
    <li>vimPlugins.zoxide-vim</li>
    <li>yazi</li>
    <li>zoxide</li>
  </ul>
</details>

**Review by cole-h — APPROVED** [review]
Upstream diff LGTM https://github.com/ajeetdsouza/zoxide/compare/v0.9.2...3022cf3686b85288e6fbecb2bd23ad113fd83f3b

Its README calls out that they won't support anything less than Nushell 0.86.0, but the `23.11-beta` tag has 0.87.0, so this is good, and also should probably be backported to 23.11 to fix this there as well.

**Pull request merged**